### PR TITLE
[4.4.0] Fix compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
See #24124 for the gcc ones (already merged)

For the MSVC one: @kbloom, @cbjeukendrup  please check!

### CODE REVIEW

File: src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ void PopupWindow_QQuickView::init(QQmlEngine* engine, bool isDialogMode, bool is
     // popup
     else {
         Qt::WindowFlags flags(
-#ifdef defined(Q_OS_WIN) || defined(Q_OS_MACOS)
+#if defined(Q_OS_WIN) || defined(Q_OS_MACOS)
```
[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp:78]: master has it simply as
```cpp
        Qt::WindowFlags flags(
#ifdef Q_OS_LINUX
            Qt::Popup // Popups can't be Qt::Tool on Linux Wayland, or they can't be relatvely positioned.
#else
            Qt::Tool
#endif
            | Qt::FramelessWindowHint            // Without border
```
So something in the backporting went wrong

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp:78]: Functionally 4.4.0 and master are now the same, (except for platforms that are neither Linux, macOS or Windows (*BSD anyone?), code-wise they differ theough.

[the remark this answers is not in the document]
Comment (cbjeukendrup) [reply src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp:78]: Yes, BSD was the reason that I chose to do it this way (but of course it's a shame that I made a typo...)

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp:78]: So now that change is needed for master too, right?

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp:78]: But  does *BSD also have that Wayland issue?

[the remark this answers is not in the document]
Comment (kbloom) [reply src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp:78]: BSD should probably be treated the same as Linux.

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp:78]: OK, it is now in 4.4.0, but not (yet) in master

[the remark this answers is not in the document]
Comment (Jojo-Schmitz) [reply src/framework/uicomponents/view/popupwindow/popupwindow_qquickview.cpp:78]: See #24219

**Pull request merged**